src/components/organisms/CompareModal/ComparisonList.tsx

```diff
@@ -15,10 +15,25 @@ import {ComparisonItemProps, HeaderItemProps} from './types';
 
 export const ComparisonList: React.FC = () => {
   const items = useAppSelector(state => selectComparisonListItems(state.compare));
+  const search = useAppSelector(state => state.compare.current.search);
+
+  const searchedItems = useMemo(() => {
+    if (!search) return items;
+    return items
+      .filter(item => {
+        if (item.type === 'header') return true;
+        return item.name.toLowerCase().includes(search);
+      })
+      .filter((item, index, partiallySearchedItems) => {
+        if (item.type === 'comparison') return true;
+        const nextItem = partiallySearchedItems[index + 1];
+        return nextItem && nextItem.type === 'comparison';
+      });
+  }, [items, search]);
 
   return (
     <div>
-      {items.map(item => {
+      {searchedItems.map(item => {
```

Comment:
Filtering on search so late in the process makes this performant. Compared to the upcoming filters you no longer have to do any recalculations. The downside of this approach is that the headers keep showing the total resource count instead the filtered resource count. I think this is a decent trade-off but I wanted to highlight this as it could be considered a bug. What do you think?